kqueen_ui/blueprints/ui/templates/ui/partial/_cluster_row.html

```diff
@@ -1,44 +1,58 @@
-            {% if cluster.metadata %}
-            <a data-toggle="collapse" data-target="#cluster-{{ index }}" class="clickable" title="More information">
-              <i class="mdi mdi-information-outline"></i>
-            </a>
-            {% endif %}
-            {{ render_cluster_row_actions(cluster) }}
-          </td>
-        </tr>
-        {% if cluster.metadata %}
-        <tr id="metadata-cluster-{{ index }}">
-          <td colspan="5" style="padding: 0px; border-top: 0px;">
-            <div style="margin: 8px;" id="cluster-{{ index }}" class="collapse">
-              <table class="table table-hover no-actions" id="table-clustermeta">
-                <tbody>
-                  {% for key, value in cluster.metadata.items() %}
-                  <tr>
-                    <th style="width: 300px;">{{ key }}</th>
-                    <td>{{ value|urlize(40, target="_blank") }}</td>
-                  </tr>
-                  {% endfor %}
-                </tbody>
-              </table>
-            </div>
-          </td>
-        </tr>
-        {% endif %}
+  <tr
+    {% if cluster.state in config.CLUSTER_TRANSIENT_STATES %}
+      class="in-transition"
+      data-href="{{ url_for('ui.cluster_row', cluster_id=cluster.id, index=index) }}"
+    {% endif %}
+    class="clickable"
+    data-index="{{ index }}"
+    data-status="{{ cluster.state }}"
+    data-href="{{ url_for('ui.cluster_detail', cluster_id=cluster.id) }}"
+  >
+    <td class="table-col-4">
+      {% if is_authorized(session, 'cluster:get', cluster) %}
+      <div class="cluster-name">
+        <a href="{{ url_for('ui.cluster_detail', cluster_id=cluster.id) }}">{{ cluster.name }}</a>
+        <a href="{{ detail_url }}" title="Rename" class="cluster-rename">
+          <i class="mdi mdi-pencil"></i>
+        </a>
+      </div>
+      {% else %}
+      {{ cluster.name }}
+      {% endif %}
+    </td>
+    <td class="table-col-4">{{ cluster.provisioner.name }}</td>
+    <td class="table-col-4">{{ cluster.created_at }}</td>
+    <td class="status_column">
+      {{ cluster.state|cluster_status_icon|safe }}
+    </td>
+    <td class="action_column">
+      {% if cluster.metadata %}
+      <a data-toggle="collapse" data-target="#cluster-{{ index }}" class="clickable" title="More information">
+        <i class="mdi mdi-information-outline"></i>
+      </a>
+      {% endif %}
+      {{ render_cluster_row_actions(cluster) }}
+    </td>
+  </tr>
+  {% if cluster.metadata %}
+  <tr id="metadata-cluster-{{ index }}">
+    <td colspan="5" style="padding: 0px; border-top: 0px;">
+      <div style="margin: 8px;" id="cluster-{{ index }}" class="collapse">
+        <table class="table table-hover no-actions" id="table-clustermeta">
+          <tbody>
+            {% for key, value in cluster.metadata.items() %}
+            <tr>
+              <th style="width: 300px;">{{ key }}</th>
+              <td>{{ value|urlize(40, target="_blank") }}</td>
```

Comment:
value can be list or dictionary as well. Could you please resolve this case also?)

Reply:
Only if you allow me to refactor quotes, indentation and line breaks in the same commits with the main request  :/

Reply:
I thought it is new code, it is hard to catch what is changed in _cluster_row.html due to refactoring again. I'm not able to review it